packages/common/src/types/rejectReason.ts

```diff
@@ -0,0 +1,202 @@
+    | RejectReasonTag.BakerInCooldown
+    | RejectReasonTag.NonExistentCredentialID
+    | RejectReasonTag.KeyIndexAlreadyInUse
+    | RejectReasonTag.InvalidAccountThreshold
+    | RejectReasonTag.InvalidCredentialKeySignThreshold
+    | RejectReasonTag.InvalidEncryptedAmountTransferProof
+    | RejectReasonTag.InvalidTransferToPublicProof
+    | RejectReasonTag.InvalidIndexOnEncryptedTransfer
+    | RejectReasonTag.ZeroScheduledAmount
+    | RejectReasonTag.NonIncreasingSchedule
+    | RejectReasonTag.FirstScheduledReleaseExpired
+    | RejectReasonTag.InvalidCredentials
+    | RejectReasonTag.RemoveFirstCredential
+    | RejectReasonTag.CredentialHolderDidNotSign
+    | RejectReasonTag.NotAllowedMultipleCredentials
+    | RejectReasonTag.NotAllowedToReceiveEncrypted
+    | RejectReasonTag.NotAllowedToHandleEncrypted
+    | RejectReasonTag.MissingBakerAddParameters
+    | RejectReasonTag.FinalizationRewardCommissionNotInRange
+    | RejectReasonTag.BakingRewardCommissionNotInRange
+    | RejectReasonTag.TransactionFeeCommissionNotInRange
+    | RejectReasonTag.AlreadyADelegator
+    | RejectReasonTag.InsufficientBalanceForDelegationStake
+    | RejectReasonTag.MissingDelegationAddParameters
+    | RejectReasonTag.InsufficientDelegationStake
+    | RejectReasonTag.DelegatorInCooldown
+    | RejectReasonTag.StakeOverMaximumThresholdForPool
+    | RejectReasonTag.PoolWouldBecomeOverDelegated
+    | RejectReasonTag.PoolClosed;
+
+export type ModuleRefRejectReasonTag =
+    | RejectReasonTag.ModuleHashAlreadyExists
+    | RejectReasonTag.InvalidModuleReference;
+
+export type AccountAddressRejectReasonTag =
+    | RejectReasonTag.InvalidAccountReference
+    | RejectReasonTag.NotADelegator
+    | RejectReasonTag.NonExistentRewardAccount
+    | RejectReasonTag.NotABaker
+    | RejectReasonTag.ScheduledSelfTransfer
+    | RejectReasonTag.EncryptedAmountSelfTransfer;
+
+export type StringRejectReasonTag =
+    | ModuleRefRejectReasonTag
+    | AccountAddressRejectReasonTag
+    | RejectReasonTag.NonExistentCredentialID
+    | RejectReasonTag.DuplicateAggregationKey;
+
+export interface StringRejectReason {
+    tag: StringRejectReasonTag;
+    contents: string;
+}
+export type NumberRejectReasonTag =
+    | RejectReasonTag.AlreadyABaker
+    | RejectReasonTag.DelegationTargetNotABaker;
+
+export interface NumberRejectReason {
+    tag: NumberRejectReasonTag;
+    contents: number;
+}
+
+export interface SimpleRejectReason {
+    tag: SimpleRejectReasonTag;
+}
+
+export interface InvalidReceiveMethod {
+    tag: RejectReasonTag.InvalidReceiveMethod;
+    contents: {
+        moduleRef: string;
+        receiveName: string;
+    };
+}
+
+export interface InvalidInitMethod {
+    tag: RejectReasonTag.InvalidInitMethod;
+    contents: {
+        moduleRef: string;
+        initName: string;
+    };
+}
+
+export interface AmountTooLarge {
+    tag: RejectReasonTag.AmountTooLarge;
+    contents: {
+        address: string;
+        amount: string;
```

Comment:
Not bigint?

Reply:
I'll test it.

Reply:
![Screenshot from 2023-01-10 13-22-31](https://user-images.githubusercontent.com/87635671/211550812-e454b524-a105-4aeb-b83a-5ca1370d0348.png)

It should be a string. but the address is also a Address it seems, and contents is an array :upside_down_face:

Reply:
Are we sure the other contents are not also lists?

Reply:
No, I suspected the same, and they are also lists.